doc/shared/sunadjoint/SUNAdjoint_Introduction.rst

```diff
@@ -0,0 +1,120 @@
+.. ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2025, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNAdjoint.Introduction:
+
+Introduction to Adjoint Sensitivity Analysis
+============================================
+
+This section presents the :c:type:`SUNAdjointStepper` and
+:c:type:`SUNAdjointCheckpointScheme` classes. The :c:type:`SUNAdjointStepper`
+represents a generic adjoint sensitivity analysis (ASA) procedure to obtain the adjoint
+sensitivities of an IVP of the form
+
+.. math::
+   \dot{y}(t) = f(t, y, p), \qquad y(t_0) = y_0, \qquad y \in \mathbb{R}^N,
+   :label: SUNADJOINT_IVP
+
+where :math:`p` is some set of :math:`N_s` problem parameters.
+
+.. note::
+  The API itself does not implement ASA, but it provides a common
+  interface for ASA capabilities implemented in the SUNDIALS packages. Right now it supports :ref:`the
+  ASA capabilities in ARKODE <ARKODE.Mathematics.ASA>`, while the ASA capabilities in :ref:`CVODES
+  <CVODES.Mathematics.ASA>` and :ref:`IDAS <IDAS.Mathematics.ASA>` must be used directly.
+
+Suppose we have a functional :math:`g(t_f, y(t_f), p)` for which we would like to compute the gradients
+:math:`dg(t_f, y(t_f), p)/dy(t_0)` and/or :math:`dg(t_f, y(t_f), p)/dp`.
+This most often arises in the form of an optimization problem such as
+
+.. math::
+   \min_{y(t_0), p} g(t_f, y(t_f), p)
+   :label: SUNADJOINT_OPTIMIZATION_PROBLEM
+
+
+.. warning::
+   The CVODES and IDAS documentation use :math:`\lambda` to represent the adjoint variables needed
+   to obtain the gradient :math:`dG/dp` where :math:`G` is an integral of :math:`g`.
+   Our use of :math:`\lambda` in the following is akin to the use of :math:`\mu` in the CVODES and
+   IDAS docs.
+
+The adjoint method is one approach to obtaining the gradients that is particularly efficient when
+there are relatively few functionals and a large number of parameters. While :ref:`CVODES
+<CVODES.Mathematics.ASA>` and :ref:`IDAS <IDAS.Mathematics.ASA>` *continuous* adjoint methods
+(differentiate-then-discretize), ARKODE provides *discrete* adjoint methods
+(discretize-then-differentiate). For the continuous approach, we derive and solve the adjoint IVP
+backwards in time
+
+.. math::
+   \dot{\lambda}(t) = -f_y^*(t, y, p) \lambda,\quad \lambda(t_F) = g_y^*(t_f, y(t_f), p)
```

Comment:
Was this minus the source of the differences between ARKODE and CVODES in the other PR? Should we have the user supply `-f_y^*(t, y, p)` or `f_y^*(t, y, p)`?